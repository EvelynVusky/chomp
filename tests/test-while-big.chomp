int main() {
    bool one_found = false;
    word w = {{0000000000010000}};
    int size_word = 16;
    int idx = 0;
    int idx_of_one = 0;

    while ((one_found == false) && (idx < size_word)) {
        if (getBit(w, idx) == {{1}}){
            println("found a 1 in the bin type");
            one_found = true;
            idx_of_one = idx;
        }
        idx = idx + 1;
    }

    if (one_found != false){
        w = flipBit(w, idx_of_one);
    }

    println(w);
    return 0;
}