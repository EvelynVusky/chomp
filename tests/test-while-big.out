found a 1 in the bin type
0000000000000000
